src/main/kotlin/racingCar/RacingCarGame.kt

```diff
@@ -2,32 +2,49 @@ package racingCar
 
 import racingCar.entity.Car
 import racingCar.entity.CarImpl
+import racingCar.entity.CarImpl.Companion.MOVE_END_POINT
+import racingCar.entity.CarImpl.Companion.MOVE_START_POINT
 import racingCar.ui.Input
 import racingCar.ui.Result
 
 class RacingCarGame {
     fun play() {
-        var cars = generateCars(Input().getNumberOfCars())
-        val moves = Input().getNumberOfMoves()
+        val cars = generateCars(Input.getCarNames())
+        val moves = Input.getNumberOfMoves()
+        val movedCars = getMovedCarList(cars, moves)
+        Result.printWinner(movedCars)
+    }
+
+    fun getMovedCarList(cars: List<Car>, moves: Int): List<Car>{
+        val movedCars = cars.toMutableList()
         repeat(moves) {
-            cars = moveCars(cars)
+            repeat(cars.size) {
+                val car = movedCars.removeFirst()
+                movedCars.add(moveCar(car))
+            }
+            Result.printCarsDistanc(movedCars)
         }
+        return movedCars
```

Comment:
```suggestion
        return movedCars.toList()
```
외부에 값을 전달할 때에는, 외부에서 원본을 손상시키지 않도록 방어적 복사를 통해서 전달해주는 것이 좋아요.